### DISCUSSION TITLE
Can you configure collection style for maps and sequences separately?

### DISCUSSION THREAD

**Discussion by marcusolsson** [eemeli/yaml#488] [Q&A]
I want to configure the collection style for maps and sequences separately. Specifically, `block` for maps and `flow` for sequences.

It looks like I could set the `flow` option on the `CreateNode` options, but I'm not sure how I'd use it with the `stringify` API.

**Comment by slingshotsys** [comment]
I know this is old, but I have the same question.  Did you ever figure this out?

**Comment by eemeli** [comment]
There's no built-in option for this, but if you first build a `Document` from your JS value, you can use a visitor to set `node.flow = true` on all sequences, and only then serialize the document: https://eemeli.org/yaml/#finding-and-modifying-nodes

**Reply by slingshotsys** [reply]
Perfect, thanks!

In case it's helpful to anyone else, the following worked for me:

```typescript
// load the data
const doc = new YAML.Document(data);
// display sequences of strings inline (flow) but keep Maps and nested or more elaborate sequences as blocks
YAML.visit(doc, {
  Seq(key, node) {
	  node.flow = node.items.every((item) => typeof item.value === 'string');
  },
});

const yaml_string = doc.toString();
```
